### DISCUSSION TITLE
Changes is Editor

### DISCUSSION THREAD

**Discussion by adilshaikh19** [massCodeIO/massCode#495] [Ideas]
In the section where we add or edit code, could we consider switching the editor to something like Monaco Editor or another advanced code editor? This would provide better flexibility with features like syntax highlighting, code suggestions, and searching within snippets.

**Comment by antonreshetov** [comment]
Not planned. Originally there in v1 was Monaco, but it was very difficult to find enough grammars for it, so Codemirror was chosen in version 2

**Reply by pa-0** [reply]
> Not planned. Originally there in v1 was Monaco, but it was very difficult to find enough grammars for it, so Codemirror was chosen in version 2

That is surprising. Between Codemirror and Monaco, I would have guessed that Monaco would have way more community support, features, etc.  I mean you would think that the fact that it is being developed by M$ _and_ integrated into VS Code would have the opposite effect.  Then again,  circumstances were probably much different when you were developing v1

**Reply by antonreshetov** [reply]
In general, for a simple snippet manager there is no need for a full-fledged code editor, you just need syntax highlighting. A full-fledged editor is quite a heavy dependency that will increase the already heavy weight associated with Electron. This is one of the reasons why I chose Codemirror.

**Comment by adilshaikh19** [comment]
Okay, can we add search or find functionality inside a code editor.
